DeepEqual.js

```diff
@@ -0,0 +1,67 @@
+// Глубокое сравнение
+
+const deepEqual = (first, second) => {
+    if (typeof first == 'object' && typeof second == 'object') {
+
+        // Проверяем, не null ли какой-то из сравниваемых обьектов
+        if (first == null || second == null) {
+            return false;
+        }
+        // Сравниванием по длинне. Если в каком-то аргументов больше, они уже не равны
+        if (first.length != second.length) {
+            return false;
+        }
+
+        // Проходим по свойствам первого обьекта
+        for (let key in first) {
+
+            // Проверяем, есть ли у второго обьекта такое же свойство, как у первого
+            if (first.hasOwnProperty(key) !== second.hasOwnProperty(key)) {
+                return false;
+            }
+
+            // Смотрим тип свойства
+            switch (typeof first[key]) {
+                case 'object':
+                    if (!deepEqual(first[key], second[key])) {
+                        return false;
+                    }
+                    break;
+                case 'function':
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Varlyle/my-first-project/pullRequest?prid=7306920)
```suggestion
                case "function":
```